File: LeetCode/258. Add Digits/elixir/solution.exs
defmodule Solution do
  @spec add_digits(num :: integer) :: integer
  def add_digits(num) do
    num
    |> to_digits()
    |> sum_digits()
  end

  defp to_digits(num) do
    num |> to_string() |> String.codepoints() |> Enum.map(&String.to_integer/1)
  end

  defp sum_digits([num]) do
    num
  end

  defp sum_digits(nums) do
    nums |> Enum.sum() |> to_digits() |> sum_digits()
  end
end

File: LeetCode/258. Add Digits/java/Solution.java
class Solution {
    public int addDigits(int num) {
        if (num < 10) {
            return num;
        }
        int sum = Arrays.stream(toDigits(num)).sum();
        return addDigits(sum);
    }

    private int[] toDigits(int number) {
        String numberStr = Integer.toString(number);
        int[] digits = new int[numberStr.length()];
        for (int i = 0; i < numberStr.length(); i++) {
            digits[i] = numberStr.charAt(i) - '0';
        }
        return digits;
    }
}